django/forms/formsets.py

```diff
@@ -139,9 +140,16 @@ def forms(self):
         Instantiate forms at first property access.
         """
         # DoS protection is included in total_form_count()
-        forms = [self._construct_form(i) for i in range(self.total_form_count())]
+        forms = [self._construct_form(i, **self.get_form_kwargs(i))
+                 for i in range(self.total_form_count())]
         return forms
 
+    def get_form_kwargs(self, index):
```

Comment:
Actually... I might just merge this as is, and clean up a couple of docs problems once it's in trunk.